modules/op-about-whenexpression.adoc

```diff
@@ -0,0 +1,155 @@
+// This module is included in the following assembly:
+//
+// *openshift_pipelines/understanding-openshift-pipelines.adoc
+
+[id="about-whenexpression_{context}"]
+= When expression
+
+When expressions guard task execution by setting criteria for the execution of tasks within a pipeline. They contain a list of components that allows a task to run only when certain criteria are met. When expressions are supported in the final set of tasks that are specified using the `finally` field in the pipeline YAML file. When expressions in the `finally` task provide efficient guarded execution and improved reusability of tasks.
```

Comment:
Reading "when" like this feels strange. My brain is constantly asking "Which when am I looking at?" :-)
Would it help to put `when` between backticks to signify that it is a code element? 
```suggestion
`when` expressions guard task execution by setting criteria for the execution of tasks within a pipeline. They contain a list of components that allows a task to run only when certain criteria are met. `when` expressions are supported in the final set of tasks that are specified using the `finally` field in the pipeline YAML file. `when` expressions in the `finally` task provide efficient guarded execution and improved reusability of tasks.
```

Reply:
I agree. Thanks for bringing up this question. But here in the pipeline context, the users are aware of the when expressions and what they mean like in the running content unless we are talking about this from the specification context. Hence retaining as is.